Feature/debug flag

kees2125 opened the pull request:
adds the debug flag option to DashDashVersion so a version for a debug build can be created.
DIVDEV-1273

kees2125 committed:
removes the requirement for a core-version tag on the git repository and updates the readme

kees2125 committed:
Adds the debug-flag to generate debug version number

kees2125 committed:
adds a test for the new feature

kees2125 committed:
Merge branch 'develop' into feature/debug-flag

edsonaf commented on src/DashDashVersion/VersionNumber.cs:
```diff
@@ -142,6 +146,11 @@ public int CompareTo(VersionNumber other)
         public override string ToString() => FullSemVer;
 
+        /// <summary>
+        /// This property convaise whether the debug flag should be added to the version number.
```
convaise = conveys

kees2125 committed:
increases the patch version if a debug build is made on a core-version

edsonaf commented on src/DashDashVersion/VersionNumber.cs:
```diff
@@ -142,6 +146,11 @@ public int CompareTo(VersionNumber other)
+        /// <summary>
+        /// This property convaise whether the debug flag should be added to the version number.
+        /// </summary>
+        public bool debugVersion;
```
🙃 a property should have a getter and/or setter. and preferably first word capitalized.
if it was only used internally, I would have suggested to use a private field, but it is also being used publicly, which is why I suggest to make it a auto property.

edsonaf replied:
🔧 alternatively, you could also create a small function to set the debugVersion and set the debugVersion field private and only use it internally

boekabart commented:
Please add example values to issue.

edsonaf commented on src/git-flow-version/Program.cs:
```diff
@@ -35,6 +35,7 @@ private static int Main(string[] args)
 
             var optionBranch = app.Option("-b|--branch", "Manually set the branch to use, for determining the branch 'type' and pre-release label. This can be a full or partial name.", CommandOptionType.SingleValue);
             var optionVersion = app.Option("-v|--version", "Returns the currently installed version of git-flow-version.", CommandOptionType.NoValue);
+            var optionDebug = app.Option("-d|--debug", "Adds a debug prerelease lable to the version number.", CommandOptionType.NoValue);
```
label

kees2125 committed:
refactoring based on feedback

kees2125 committed:
refactor based on feedback

kees2125 committed:
Merge branch 'feature/NoCoreVersionSupport' into feature/debug-flag

kees2125 committed:
Changes the Core debug version and fixes a few issues

Core debug versions are now 1.0.1-debug.1.0.0.
Protects the feature ordinal from overflowing by throwing a exception instead.
Git commits are now orderd by parent in the main branch instead of by commit time as to avoid inconsistent ordering.
The feature ordinal is now calculated by counting the unique commits on the feature branch compaired to develop.

kees2125 committed:
dotnet format

kees2125 committed:
fixes the debug flag and refactors the code

Commit:
Move code from helper methods to GitBranch

kees2125 committed:
Adds a check to make sure core-version tags are only on master to prevent user error or remote reffs not being up to date.

kees2125 committed:
Update dotnet sdk to the latest in azure build pipeline

kees2125 committed:
Fixes broken test that is no longer valid with the new version

basbossink reviewed (approved):
👍

boekabart commented on src/DashDashVersion/DashDashVersion.csproj:
```diff
@@ -1,33 +1,36 @@
-    <PackageReference Include="System.Linq" Version="4.3.0" />
-  </ItemGroup>
-</Project>
+    <Compile Remove="..\..\built\SharedAssemblyInfo.cs" />
```
Was this intentional?

kees2125 replied:
No thank you, I hadn't noticed.

boekabart commented on src/DashDashVersion/RepositoryAbstraction/GitBranch.cs:
```diff
@@ -26,18 +26,25 @@ namespace DashDashVersion.RepositoryAbstraction
             string friendlyName,
             bool isCurrentRepositoryHead,
-            IEnumerable<GitCommit> commits)
+            List<GitCommit> commits)
```
🔧 I think the minimal interface to use here is `IReadonlyCollection`

boekabart commented on src/DashDashVersion/RepositoryAbstraction/GitRepoReader.cs:
```diff
@@ -240,19 +219,11 @@ private GitBranch FindBranch(string branchName)
                 FindBranch(branchName);
 
-        private List<GitTag> VisibleTags() => 
+        private List<GitTag> VisibleTags() =>
```
🔧 property instead of method

kees2125 committed:
Updated dependencies and removed a weird artefact from the DashDashVersion.csproj

kees2125 committed:
Changed the collection type to the most generic acceptable type

kees2125 committed:
Changes methods into properties

kees2125 committed:
Changes the debug flag on a version number to just have a get and set

kees2125 committed:
file mist in the last commit for the debug flag on version number
